distributed/utils.py

```diff
@@ -482,6 +484,47 @@ def join(self, timeout: float | None = None) -> None:
                 del self._exception
 
 
+def run_and_close_tornado(
+    async_fn: Callable[P, Coroutine[AnyType, AnyType, T]],
+    /,
+    *args: P.args,
+    **kwargs: P.kwargs,
+) -> T:
+    tornado_loop = None
+
+    async def inner_fn():
+        nonlocal tornado_loop
+        tornado_loop = IOLoop.current()
+        return await async_fn(*args, **kwargs)
+
+    try:
+        return asyncio.run(inner_fn())
+    finally:
+        assert tornado_loop is not None
+        tornado_loop.close(all_fds=True)
+
+
+# distributed.Client objects are closed using an atexit hook that calls
+# Client.close, this sends an {"op": "close-client"} to the scheduler.
+# On the Windows ProactorEventLoop tornado also sets an atexit hook that
+# closes the AddThreadSelectorEventLoop that is managing our Client's socket
+# to the scheduler causing the {"op": "close-client"} message to fail.
+# because we manage the cleanup of the tornado event loop ourselves we want to
+# disable the tornado atexit hook
```

Comment:
ah this isn't quite right it's actually LocalCluster.close calling await self.scheduler_comm.retire_workers